llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -3107,14 +3107,14 @@ bool LoopVectorizationCostModel::isPredicatedInst(Instruction *I) const {
     // is correct.  The easiest form of the later is to require that all values
     // stored are the same.
     return !(Legal->isInvariant(getLoadStorePointerOperand(I)) &&
-             TheLoop->isLoopInvariant(cast<StoreInst>(I)->getValueOperand()));
+             Legal->isInvariant(cast<StoreInst>(I)->getValueOperand()));
```

Comment:
This change caused a subtle mis-compile: https://github.com/llvm/llvm-project/issues/149347

For now it is probably best to undo this change: https://github.com/llvm/llvm-project/pull/150828